api/query/cache/index/index_test.go

```diff
@@ -148,5 +151,126 @@ func TestEvictNonEmpty(t *testing.T) {
 	assert.Nil(t, i.EvictAnyRun())
 }
 
+func TestSync(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	loader := NewMockReportLoader(ctrl)
+	i, err := NewShardedWPTIndex(loader, 1)
+	assert.Nil(t, err)
+
+	loader.EXPECT().Load(gomock.Any()).DoAndReturn(func(run shared.TestRun) (*metrics.TestResultsReport, error) {
+		strID := strconv.FormatInt(run.ID, 10)
+		strStatus := shared.TestStatusStringFromValue(run.ID % 7)
+		return &metrics.TestResultsReport{
+			Results: []*metrics.TestResults{
+				&metrics.TestResults{
+					Test:   "shared",
+					Status: strStatus,
+				},
+				&metrics.TestResults{
+					Test:   "test" + strID,
+					Status: "PASS",
+				},
+			},
+		}, nil
+	}).AnyTimes()
+
+	i.IngestRun(makeRun(1))
+	i.IngestRun(makeRun(2))
+	i.IngestRun(makeRun(3))
+	i.IngestRun(makeRun(4))
+	i.IngestRun(makeRun(5))
+	i.IngestRun(makeRun(6))
+	i.IngestRun(makeRun(7))
+	i.IngestRun(makeRun(8))
+
+	var wg sync.WaitGroup
+	for j := 9; j <= 16; j++ {
```

Comment:
Some explanatory comments are needed for what this is actually doing async and what failure would look like.

Reply:
Added comments. @lukebjerring PTAL again.